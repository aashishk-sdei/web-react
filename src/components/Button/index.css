/* Common */
.btn {
  @apply font-medium py-1.5 px-4 rounded-lg border border-transparent focus:outline-blue  flex items-center;
}

/* Button Type */
.btn-default {
  @apply bg-gray-1 text-gray-7 hover:bg-gray-2;
}

.btn-default-dark {
  @apply bg-gray-2 text-blue-4 hover:bg-gray-3;
}

.btn-primary-inverted {
  @apply bg-gray-1 text-blue-4 hover:bg-gray-2;
}

.btn-primary {
  @apply bg-blue-4 text-white hover:bg-blue-5;
}

.btn-secondary {
  @apply bg-gray-7 text-white hover:bg-black;
}

.btn-red {
  @apply bg-redShades-4 border-transparent text-white hover:bg-redShades-5;
}
.btn-red-dark {
  @apply bg-redShades-6 border-transparent text-white hover:bg-redShades-7;
}

.btn-danger {
  @apply bg-maroon-5 text-white hover:bg-maroon-6;
}
.btn-success {
  @apply bg-green-4 text-white hover:bg-green-6;
}
.btn-skyblue {
  @apply bg-skyblue-6 border-skyblue-6 text-white hover:bg-skyblue-7 hover:border-skyblue-7;
}
.btn-white {
  @apply bg-white border-white text-redShades-7 hover:bg-gray-2 hover:border-gray-2 hover:text-redShades-7;
}
.link-white {
  @apply px-3 bg-transparent border-transparent text-white hover:bg-transparent hover:border-transparent hover:text-gray-2;
}
.link-white .icon-large {
  @apply mt-auto mb-auto;
}
.link-white .icon-large svg {
  @apply w-4;
}
.link-white .swapper {
  @apply flex-row-reverse;
}
.link-white .ml-swapper {
  @apply ml-0 mr-2;
}
.link-danger{
  @apply bg-gray-1 text-maroon-5 hover:bg-gray-2 hover:text-maroon-6;
}
.btn-maroon {
  @apply bg-maroon-4;
}

.btn-stepper {
  @apply bg-gray-1 border border-gray-3 py-1.5 px-4 rounded-lg focus:outline-none focus:ring-2 focus:ring-blue-5 flex items-center text-gray-4 hover:bg-blue-1 justify-center hover:text-blue-4 h-20 !important w-80;
}

/* Button Size */
.btn-small {
  @apply text-xs h-6;
}

.btn-medium {
  @apply text-sm h-8;
}

.btn-large {
  @apply text-base h-10;
}
.btn-xlarge {
  @apply text-2xl h-16 md:h-21.5 px-6;
}

/* Button Disable */
.btn-disable {
  @apply opacity-50 cursor-default;
  filter: opacity(0.8) !important;
}
.link-gray{
  @apply text-gray-7 cursor-pointer no-underline hover:text-blue-4 hover:underline;
}