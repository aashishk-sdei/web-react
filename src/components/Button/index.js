import React, { useState } from "react";
import PropTypes from "prop-types";
import "./index.css";

// Components
import Loader from "../Loader";
import Icon from "../Icon";
import Translator from "../Translator";

const Button = ({ type, title, size, width, icon, disabled, handleClick, loading, tkey, buttonType, _textColor, fontWeight, ...props }) => {
  const getBtnType = () => {
    switch (type) {
      case "default":
        return "btn-default";
      case "default-dark":
        return "btn-default-dark";
      case "primary":
        return "btn-primary";
      case "primary-inverted":
        return "btn-primary-inverted";
      case "secondary":
        return "btn-secondary";
      case "danger":
        return "btn-danger";
        case "success":
          return "btn-success";
      case "red":
        return "btn-red";
      case "red-dark":
        return "btn-red-dark";
      case "skyblue":
        return "btn-skyblue";
      case "white":
        return "btn-white";
      case "link-white":
        return "link-white";
      case "link-danger":
        return "link-danger";
      case "maroon":
        return "btn-maroon";
      case "stepper":
        return "btn-stepper";
      default:
        return "";
    }
  };
  const btnType = getBtnType();

  const getBtnSize = () => {
    switch (size) {
      case "small":
        return "btn-small";
      case "medium":
        return "btn-medium";
      case "large":
        return "btn-large";
      case "xlarge":
        return "btn-xlarge typo-font-bold";
      default:
        return "";
    }
  };
  const btnSize = getBtnSize();

  const getBtnWidth = () => {
    return (width === "full")?"w-full":""
  };
  const btnWidth = getBtnWidth();

  const btnDisable = disabled || loading ? "btn-disable" : "";

  const getlocalIconColor = () => {
    switch (type) {
      case "default":
        return "secondary";
      case "stepper":
        return "default";
      default:
        return "white";
    }
  };
  const localIconColor = getlocalIconColor();

  const [iconColor, setIconColor] = useState(localIconColor);

  const handleMouseEnter = () => {
    if (type === "stepper") setIconColor("primary");
  };

  const handleMouseLeave = () => {
    if (type === "stepper") setIconColor(localIconColor);
  };

  const getTitle = (tkeyin) => {
    return tkeyin !== "" ? <Translator tkey={tkeyin} /> : title;
  };

  const getFontWeight = () => {
    switch (fontWeight) {
      case "regular":
        return "typo-font-regular";
      case "light":
        return "typo-font-light";
      case "medium":
        return "typo-font-medium";
      case "bold":
        return "typo-font-bold";
      case "lyon-regular":
        return "lyon-font-regular";
      case "lyon-medium":
        return "lyon-font-medium";
      default:
        return "typo-font-medium";
    }
  };
  return (
      <button type={buttonType ? buttonType : "button"} className={`btn ${btnType} ${btnSize} ${btnWidth} ${btnDisable}`} onClick={!disabled ? handleClick : undefined} onMouseEnter={handleMouseEnter} onMouseLeave={handleMouseLeave} {...props}>
      {icon ? (
        <div className="flex items-center swapper">
          <Icon type={icon} color={iconColor} size={size} />
          <div className={`ml-2 typo-font-light ml-swapper ${getFontWeight()}`} > {getTitle(tkey)}</div>
        </div>
      ) : (
        <div className="flex justify-center items-center w-full">
          {loading && (
            <span className="mr-2">
              <Loader spinner={true} color="default" size={12} />
            </span>
          )}
          <span className={getFontWeight()}>{getTitle(tkey)}</span>
        </div>
      )}
    </button>
  );
};

Button.propTypes = {
  type: PropTypes.oneOf(["default", "default-dark", "primary", "primary-inverted", "secondary", "danger", "success", "skyblue", "white", "link-white","link-danger", "red", "red-dark", "stepper"]),
  title: PropTypes.string,
  size: PropTypes.oneOf(["small", "medium", "large", "xlarge"]),
  icon: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  disabled: PropTypes.bool,
  handleClick: PropTypes.func,
  tkey: PropTypes.string,
};

Button.defaultProps = {
  type: "primary",
  title: "Learn more",
  size: "medium",
  icon: false,
  disabled: false,
  handleClick: undefined,
  tkey: "",
};

export default Button;
