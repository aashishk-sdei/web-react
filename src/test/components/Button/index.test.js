import React from 'react';
import { shallow } from 'enzyme';
import Button from '../../../components/Button';


test('should test Button component with list of props', () => {
    const wrapper = shallow(
     <Button type='primary' 
     title= "Learn more" 
     size= "small" 
     icon= ""
     disabled= {false}
     handleClick= {undefined}
     />
     
    );
    expect(wrapper).toMatchSnapshot();
   });


   describe('Test Button component', () => {
    it('Test click event', () => {
      const mockCallBack = jest.fn();
  
      const button = shallow((<Button handleClick={mockCallBack}>Ok!</Button>));
      button.find('button').simulate('click');
      expect(mockCallBack.mock.calls.length).toEqual(1);
    });
  });

  