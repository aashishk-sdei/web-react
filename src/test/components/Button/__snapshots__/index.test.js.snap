// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test Button component with list of props 1`] = `ShallowWrapper {}`;
